pkg/cli/cmd/app/connections/display.go

```diff
@@ -64,71 +57,43 @@ func display(graph *applicationGraph) string {
 	}
 
 	for _, resource := range applicationResources {
-		if resource.Error != "" {
-			output.WriteString(fmt.Sprintf("Error: %s\n", resource.Error))
-			continue
-		}
-
-		output.WriteString(fmt.Sprintf("Name: %s (%s)\n", resource.Name, resource.Type))
+		output.WriteString(fmt.Sprintf("Name: %s (%s)\n", *resource.Name, *resource.Type))
 
 		if len(resource.Connections) == 0 {
 			output.WriteString("Connections: (none)\n")
 		} else {
 			output.WriteString("Connections:\n")
 			for _, connection := range resource.Connections {
-				if connection.To.Error != "" {
-					output.WriteString(fmt.Sprintf("  %s -> %s (%s)\n", connection.From.Name, "error", connection.To.Error))
+				connectionID, err := resources.Parse(*connection.ID)
+				if err != nil {
+					output.WriteString(fmt.Sprintf("Error: %s\n", err.Error()))
 					continue
 				}
 
-				// We format the connection differently depending on whether it's inbound or outbound.
-				if connection.From.ID == resource.ID {
+				connectionName := connectionID.Name()
+				connectionType := connectionID.Type()
+
+				if *connection.Direction == v20231001preview.DirectionOutbound {
 					// Outbound
-					output.WriteString(fmt.Sprintf("  %s -> %s (%s)\n", connection.From.Name, connection.To.Name, connection.To.Type))
+					output.WriteString(fmt.Sprintf("  %s -> %s (%s)\n", *resource.Name, connectionName, connectionType))
 				} else {
 					// Inbound
-					output.WriteString(fmt.Sprintf("  %s (%s) -> %s\n", connection.From.Name, connection.From.Type, connection.To.Name))
+					output.WriteString(fmt.Sprintf("  %s (%s) -> %s\n", connectionName, connectionType, *resource.Name))
 				}
 			}
 		}
 
-		if len(resource.Resources) == 0 {
+		if len(resource.OutputResources) == 0 {
 			output.WriteString("Resources: (none)\n")
 		} else {
 			output.WriteString("Resources:\n")
-			for _, resource := range resource.Resources {
-				if resource.Error != "" {
-					output.WriteString(fmt.Sprintf("Error: %s\n", resource.Error))
-					continue
-				}
-
-				link := makeHyperlink(resource)
```

Comment:
Is there a reason why this needs to be removed? I'd imagine we can still make a link for the Azure resources.

Reply:
Thanks!